src/utils/utils.ts

```diff
@@ -199,6 +181,10 @@ export function toPrimitive(value) {
     return value === null ? null : typeof value === "object" ? "" + value : value
 }
 
+export function isThennable(thing: any): thing is PromiseLike<any> {
```

Comment:
In general, it's preferable to not check if something is `then`able if suspected but to `Promise.resolve` it.

Reply:
Lol, committed this in the wrong branch.. :)